### ISSUE TITLE
Support Isabelle code generation

### ISSUE THREAD

**Issue by jonaskrask** [jonaskrask/vdm-vscode#64]
Related to issue #19

**Comment by jonaskrask** [comment]
A button has been added for translate to Isabelle on the client side (development branch) with the language id: "isabelle"

**Comment by nickbattle** [comment]
Thanks. That's fine - jars I can handle :-)

**Comment by nickbattle** [comment]
The good news is that it looks like the annotationPaths and JVMArguments settings give us the Java process settings that we need. So the folder settings.json is like this:
```
{
	"vdm-vscode.debug.experimentalServer": false,
	"vdm-vscode.highPrecision": false,
	"editor.codeLens": true,
	"vdm-vscode.annotationPaths": "/home/nick/.vscode/vdm2isa-4.4.4-SNAPSHOT-211019.jar",
	"vdm-vscode.JVMArguments": "-Dlspx.plugins=plugins.ISAPluginSL"
}
```

**Comment by nickbattle** [comment]
It's sending:
```
			"experimental": 
			{
				"translateProvider": 
				{
					"languageId": 
					[
						"latex",
						"word",
						"coverage",
						"graphviz",
						"isabelle"
					],
					"workDoneProgress": false
				},
				"dapServer": 
				{
					"port": 8001
				},
				"combinatorialTestProvider": 
				{
					"workDoneProgress": true
				},
				"proofObligationProvider": true
			}
```

**Comment by nickbattle** [comment]
Looks like you're not adding Isabelle in this constructor:
https://github.com/jonaskrask/vdm-vscode/blob/0b154cceb72a9801a641e2ccad14364accafad12/src/SpecificationLanguageClient.ts#L33

**Comment by jonaskrask** [comment]
Can we include the jar from Leo in the extension? In either case, it would be nice if the server discovers if the jar is there and only includes the isabelle translate provider as a capability if it is there. Otherwise, the button will be displayed, but won't function properly.

**Comment by nickbattle** [comment]
I'm not sure what state the Isabelle translation is in at the moment, we should check that with Leo. The second problem is that this is a plugin (server capability extension) rather than a library. We were going to come up with a way to both identify plugin jars and set the -Dlspx.plugins property in a friendly way? Lastly, the "isabelle" languageId is currently hard-wired into the Server experimental translateProvider response. Plugins can add their own response entries, but not change existing ones... which might be more flexible. Then Leo's plugin could announce itself, if it was configured.

**Comment by nickbattle** [comment]
The ISA plugin currently sets an experimental option called "isabelle" with the value "server-option" :-)  Really it ought to somehow edit or add a translateProvider entry.

**Comment by jonaskrask** [comment]
Ahh yes, that's true, forgot about that. I see, it would be nice though if plugins could contribute to the existing entries. Otherwise, plugins will have to be supported a bit awkwardly if they contribute to the existing features (e.g. a new translation).

**Comment by nickbattle** [comment]
Yes. If we pass the "standard" options to the plugins, they can edit/extend them and replace the standard ones. It should be easy enough to implement.

**Comment by nickbattle** [comment]
I've just pushed something to do this to "development". If (and only if) the ISAPlugin is on the path and -Dlspx.plugins includes it, then it will edit itself into the translateProvider language list. I've pushed a change to Leo's code to do this... so I'll email him separately to let him know he needs to pull :-)

**Comment by c4rlo14** [comment]
A vdm2isa translation plugin has been designed and implemented by @leouk. See more at https://github.com/leouk/VDM_Toolkit/tree/main/vdm2isa
